influxdb3_write/src/lib.rs

```diff
@@ -136,21 +165,18 @@ pub trait Persister: Debug + Send + Sync + 'static {
     fn object_store(&self) -> Arc<dyn object_store::ObjectStore>;
 }
 
-#[async_trait]
```

Comment:
I see that here and a few places operations on the WAL were moved from sync over  to async What was the primary motivation for that change?

Reply:
There's no need for the WAL operations to be async. They in fact should be blocking synchronous operations. All of the actual wal functions beyond initial loadup will be run inside a separate spawned task/thread. The Buffer will provide the async API, which the server will use.